app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/catalog/ui/view/ThirdPageContent.kt

```diff
@@ -0,0 +1,347 @@
+        linkTo(start = parent.start, end = parent.end)
+        top.linkTo(if (isFeatureHorizontal) horizontalGuideline else firstRowRef.bottom, topMargin)
+    }
+}
+
+@Composable
+private fun getConstraintSetForSmallWindowWidth() = ConstraintSet {
+    val firstImageRef = createRefFor(THIRD_PAGE_FIRST_IMAGE_ID)
+    val firstTextRef = createRefFor(THIRD_PAGE_FIRST_TEXT_ID)
+    val secondImageRef = createRefFor(THIRD_PAGE_SECOND_IMAGE_ID)
+    val secondTextRef = createRefFor(THIRD_PAGE_SECOND_TEXT_ID)
+    val thirdTextRef = createRefFor(THIRD_PAGE_THIRD_TEXT_ID)
+
+    val horizontalMargin = 36.dp
+    val topMargin = 20.dp
+    val smallMargin = 8.dp
+
+    constrain(firstImageRef) {
+        top.linkTo(parent.top, topMargin)
+        linkTo(start = parent.start, end = parent.end)
+    }
+
+    constrain(firstTextRef) {
+        top.linkTo(firstImageRef.bottom, topMargin)
+        linkTo(
+            start = parent.start,
+            end = parent.end,
+            startMargin = horizontalMargin,
+            endMargin = horizontalMargin
+        )
+    }
+
+    constrain(secondImageRef) {
+        top.linkTo(firstTextRef.bottom, smallMargin)
+        linkTo(start = parent.start, end = parent.end)
+    }
+
+    constrain(secondTextRef) {
+        top.linkTo(secondImageRef.bottom, topMargin)
+        start.linkTo(parent.start, horizontalMargin)
+    }
+
+    constrain(thirdTextRef) {
+        top.linkTo(secondTextRef.bottom)
+        linkTo(
+            start = parent.start,
+            end = parent.end,
+            startMargin = horizontalMargin,
+            endMargin = horizontalMargin
+        )
+    }
+}
+
+@Composable
+fun ThirdPageContent(
+    modifier: Modifier,
+    constraintSet: ConstraintSet,
+    catalogItem: CatalogItem,
+    isFeatureHorizontal: Boolean,
+    isSinglePortrait: Boolean,
+    showTwoPages: Boolean,
+    isFoldStateHalfOpened: Boolean
+) {
+    ConstraintLayout(
+        constraintSet = constraintSet,
+        modifier = modifier
+    ) {
+        Row(
+            modifier = Modifier
+                .padding(
+                    top = dimensionResource(id = R.dimen.catalog_top_margin),
+                    start = dimensionResource(id = R.dimen.catalog_horizontal_margin),
+                    end = dimensionResource(id = R.dimen.normal_margin)
+                )
+                .layoutId(THIRD_PAGE_FIRST_ROW_ID),
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+
+            RoundedImage(
+                modifier = Modifier
+                    .weight(ROW_WEIGHT)
+                    .requiredWidth(
+                        if ((isFeatureHorizontal && !showTwoPages) ||
+                            isSinglePortrait || (isFoldStateHalfOpened && !showTwoPages)
+                        )
+                            dimensionResource(id = R.dimen.catalog_max_image_height) else
+                            dimensionResource(id = R.dimen.catalog_min_image_width)
```

Comment:
just a small codestyle comment, but I think usually it's more readable to have the else start on a new line when the if/else takes up more than one line (this applies to several other places in this file)
```suggestion
                            dimensionResource(id = R.dimen.catalog_max_image_height)
                        else
                            dimensionResource(id = R.dimen.catalog_min_image_width)
```